dht.go

```diff
@@ -411,7 +415,15 @@ func (dht *IpfsDHT) putLocal(key string, rec *recpb.Record) error {
 // supports the DHT protocol.
 func (dht *IpfsDHT) peerFound(ctx context.Context, p peer.ID) {
 	logger.Event(ctx, "peerFound", p)
-	dht.routingTable.HandlePeerAlive(p)
+	currentConns := dht.host.Network().ConnsToPeer(p)
+	if len(currentConns) > 0 {
+		if dht.routingTablePeerFilter(dht, currentConns) {
```

Comment:
Can we just do this in `validRTPeer`?

Reply:
this function was only being used by tests. removed.